cloud/aws/actuators/machine/actuator.go

```diff
@@ -109,7 +110,32 @@ func (a *Actuator) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 // Delete deletes a machine and is invoked by the Machine Controller
 func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	glog.Infof("Deleting machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+
+	status, err := a.machineProviderStatus(machine)
+	if err != nil {
+		return err
```

Comment:
please wrap all of these errors with `errors.Wrap` from https://godoc.org/github.com/pkg/errors

This will allow us to trace the error if we ever see it.

Something like `return errors.Wrap(err, "failed to get machine provider status")`